azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -736,32 +731,146 @@ def do_retry():
 ]
 
 
+class ReconnectState(object):
+    never_connected = "never_connected"
+    waiting_to_reconnect = "waiting_to_reconnect"
+    deferred = "deferred"
+
+
 class ReconnectStage(PipelineStage):
     def __init__(self):
         super(ReconnectStage, self).__init__()
         self.reconnect_timer = None
-        self.virtually_connected = False
+        self.state = ReconnectState.never_connected
         # connect delay is hardcoded for now.  Later, this comes from a retry policy
         self.reconnect_delay = 10
+        self.waiting_ops = []
```

Comment:
>waiting_ops [](start = 13, length = 11)

when i first read this i felt these are ops waiting for connection , but these are actually connect ops which are waiting..can this be renamed ?